xapian-letor/tests/api_letor.cc

```diff
@@ -0,0 +1,187 @@
+/** @file api_generated.cc
+ * @brief test common features of API classes
+ */
+/* Warning: This file is generated by ./generate-api_generated - do not modify directly! */
+/* Copyright (C) 2007,2009,2012,2014,2015,2016 Olly Betts
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include <config.h>
+
+#include "api_letor.h"
+
+#define XAPIAN_DEPRECATED(D) D
```

Comment:
This line is to disable deprecation warnings for features we want to test in this file.  There shouldn't be any such features in letor currently.